.github/workflows/main.yml

```diff
@@ -28,6 +28,12 @@ jobs:
   build:
     name: Build & Package
     runs-on: ubuntu-latest
+
+    # Add "id-token" with the intended permissions.
+    permissions:
+      contents: 'read'
+      id-token: 'write'
```

Comment:
Allows the job to request OIDC tokens. Required for secure authentication to cloud provider